lib/portlayer/storage/vsphere/image.go

```diff
@@ -732,3 +736,47 @@ func createBaseStructure(op trace.Operation, vmdisk *disk.VirtualDisk) (err erro
 
 	return nil
 }
+
+func (v *ImageStore) StatPath(op trace.Operation, deviceId string, target string) (stat *compute.FileStat, err error){
+
+	host, err := sys.UUID()
+	if err != nil {
+		op.Debugf("Failed to determine host UUID")
+		return nil, err
+	}
+
+	if err = v.verifyImage(op, host, deviceId); err != nil {
+		op.Debugf("Device is not an image")
+		return nil, err
+	}
+
+	// TODO: not sure if these checks are needed
```

Comment:
Others may chime in ofcourse, But I actually think these checks could be useful. It could really catch an edge case that is occurring that we may not think possible. Such as someone deleting the image store out from under us. Or somehow someone is able to get us to call the imagestore with a non existant image.  I think those could be useful checks @jzt @hickeng might also have an opinion on this as well.